File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        if(root== null) return 0;

        Queue<TreeNode> q= new LinkedList<>();
        q.add(root);
        int maxsum= Integer.MIN_VALUE;
        int level=1;
        int currlevel=0;

        while(!q.isEmpty()){
            int n=q.size();
            int sum=0;
            currlevel++;
            for(int i=0;i<n;i++){
                TreeNode curr= q.poll();
                sum+= curr.val;
                if(curr.left!= null) q.add(curr.left);
                if(curr.right!= null) q.add(curr.right);
                
            }
            if (sum > maxsum) {
                maxsum = sum;
                level = currlevel;
            }
        }
        return level;
    }
}

File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        if (root == NULL) return 0;

        queue<TreeNode*> q;
        q.push(root);

        int maxsum = INT_MIN;
        int level = 1;
        int currlevel = 0;

        while (!q.empty()) {
            int n = q.size();
            int sum = 0;
            currlevel++;

            for (int i = 0; i < n; i++) {
                TreeNode* curr = q.front();
                q.pop();

                sum += curr->val;

                if (curr->left != NULL) q.push(curr->left);
                if (curr->right != NULL) q.push(curr->right);
            }

            if (sum > maxsum) {
                maxsum = sum;
                level = currlevel;
            }
        }

        return level;
    }
};
